### ISSUE TITLE
 Specify the latest IoT SQL Rule version

### ISSUE THREAD

**Issue by ritterb-nl** [aws-samples/iot-reference-architectures#9]
In [partner-account.yaml](https://github.com/aws-samples/iot-reference-architectures/blob/master/cross-account-publish/python/partner-account.yaml), no SQL version is specified, so the default of "2015-10-08" is used. The newer "2016-03-23" version fixes some issues with top-level arrays and nested objects. Although I didn't see the SQL version documented in the [CloudFormation IoT Template](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iot-topicrule.html), the following syntax worked for me after some trial and error:
```yaml
# ...
Resources:
  # Rule to take messages from our cross account publish topic hierarchy and republish them to the appropriate customer
  CrossAccountPublishRule:
    Type: AWS::IoT::TopicRule
    Properties:
      TopicRulePayload:
        RuleDisabled: false
        AwsIotSqlVersion: 2016-03-23
        Sql: !Join [ "", [ "SELECT *, concat(\"cross_account_publish/", !Ref "AWS::AccountId", "/\", regexp_replace(topic(), \"^[^/]*/[^/]*/\", \"\")) AS topic FROM 'cross_account_publish/", !Ref CustomerAccountParameter, "/#'" ] ]
        Actions:
          - Lambda:
              FunctionArn: !GetAtt LambdaFunction.Arn
# ...
```

I've tested this by creating a CloudFormation stack with this AwsIotSqlVersion line added and verifying the SQL version of the IoT Rule. Are there any reasons not to use the latest SQL version? If not, I can create a pull request with this change.

**Comment by timmattison** [comment]
Putting the version explicitly is better than leaving it up to the CloudFormation default. That way we can be sure it will work if one region has a different default than another or if a SQL version isn’t supported there at all.

If you could, a PR would be great!

**Comment by timmattison** [comment]
Fixed in #15, thanks again!
